### PR TITLE
DEPS: drop np 1.20

### PR DESCRIPTION
**Pull request by fangchenli**

### CODE REVIEW

File: pandas/core/array_algos/putmask.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def putmask_without_repeat(
     mask : np.ndarray[bool]
     new : Any
     """
-    if np_version_under1p21:
+    if np_version_under1p22:
```
Comment (jbrockmendel) [inline pandas/core/array_algos/putmask.py:76]: is this still necessary?

File: pandas/compat/numpy/__init__.py [hunk 2]
```diff
@@ -6,17 +6,12 @@
...
 _nlv = Version(_np_version)
-np_version_under1p21 = _nlv < Version("1.21")
 np_version_under1p22 = _nlv < Version("1.22")
-np_version_gte1p22 = _nlv >= Version("1.22")
 np_version_gte1p24 = _nlv >= Version("1.24")
 is_numpy_dev = _nlv.dev is not None
-_min_numpy_ver = "1.20.3"
+_min_numpy_ver = "1.21.6"
 
-if is_numpy_dev or not np_version_under1p22:
```
Comment (jbrockmendel) [inline pandas/compat/numpy/__init__.py:16]: yah is_numpy_dev is a kludgy pattern, would be good to get rid of it

File: pandas/core/array_algos/putmask.py [hunk 3]
```diff
@@ -73,9 +72,6 @@ def putmask_without_repeat(
     mask : np.ndarray[bool]
     new : Any
     """
-    if np_version_under1p21:
-        new = setitem_datetimelike_compat(values, mask.sum(), new)
```
Comment (jbrockmendel) [inline pandas/core/array_algos/putmask.py:77]: i suspect setitem_datetimelike_compat can be deleted entirely

Comment (fangchenli) [reply pandas/core/array_algos/putmask.py:77]: There are two other usages:

https://github.com/pandas-dev/pandas/blob/249d93e4abc59639983eb3e8fccac8382592d457/pandas/core/indexes/base.py#L5327

https://github.com/pandas-dev/pandas/blob/249d93e4abc59639983eb3e8fccac8382592d457/pandas/core/internals/blocks.py#L1059

**Comment by fangchenli** [comment]
The errors seem not related. #45651

File: pandas/core/array_algos/putmask.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/array_algos/putmask.py:77]: my hope was that with the bumped version those would be unnecessary too.  i guess not?

Comment (fangchenli) [reply pandas/core/array_algos/putmask.py:77]: Removing it creates many test failures. I'll leave it for now.

Comment (jbrockmendel) [reply pandas/core/array_algos/putmask.py:77]: Darn.  Thanks for checking

**Comment by mroeschke** [comment]
1. Could you add this to the table of updated versions in the whatsnew note for 2.1?
2. Could you change this version in `pyproject.toml` and the sdist GHA workflow?
3. It looks like there's a `try/except` in `pandas/tests/extension/base/dtype.py::BaseDtypeTests:test_check_dtype` that can be removed now based on a comment regarding numpy 1.20

**Comment by mroeschke** [comment]
```
=================================== FAILURES ===================================
_______________ TestBaseDtype.test_check_dtype[decimal128(7, 3)] _______________
[gw0] linux -- Python 3.8.0 /home/runner/micromamba/envs/test/bin/python3.8
[XPASS(strict)] decimal string repr affects numpy comparison
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @fangchenli

**Pull request merged**